tests/feature-store/test_steps.py

```diff
@@ -56,3 +71,387 @@ def test_set_event_random_id():
+    data = get_data()
+    # One Hot Encode the newly defined mappings
+    one_hot_encoder_mapping = {
+        "department": list(data["department"].unique()),
+    }
+    # Define the corresponding FeatureSet
+    data_set_pandas = fs.FeatureSet(
+        "fs-new",
+        entities=[fs.Entity("id")],
+        description="feature set",
+        engine="pandas",
+    )
+    # Pre-processing grpah steps
+    data_set_pandas.graph.to(OneHotEncoder(mapping=one_hot_encoder_mapping))
+    data_set_pandas._run_db = rundb_mock
+
+    data_set_pandas.reload = unittest.mock.Mock()
+    data_set_pandas.save = unittest.mock.Mock()
+    data_set_pandas.purge_targets = unittest.mock.Mock()
+
+    # Ingest our transactions dataset through our defined pipeline
+    df_pandas = fs.ingest(
+        data_set_pandas,
+        data,
+    )
+
+    assert isinstance(df_pandas, pd.DataFrame)
+
+    # Define the corresponding FeatureSet
+    data_set = fs.FeatureSet(
+        "fs-new",
+        entities=[fs.Entity("id")],
+        description="feature set",
+    )
+    # Pre-processing grpah steps
+    data_set.graph.to(OneHotEncoder(mapping=one_hot_encoder_mapping))
+    data_set._run_db = rundb_mock
+
+    data_set.reload = unittest.mock.Mock()
+    data_set.save = unittest.mock.Mock()
+    data_set.purge_targets = unittest.mock.Mock()
+
+    # Ingest our transactions dataset through our defined pipeline
+    df = fs.ingest(
+        data_set,
+        data,
+    )
+    pd.testing.assert_frame_equal(
+        df,
+        df_pandas,
+        check_dtype=True,
+        check_index_type=True,
+        check_column_type=True,
+        check_like=True,
+        check_names=True,
+    )
+
+
+def test_pandas_step_imputer(rundb_mock):
+    data = get_data(True)
+    # Define the corresponding FeatureSet
+    data_set_pandas = fs.FeatureSet(
+        "fs-new",
+        entities=[fs.Entity("id")],
+        description="feature set",
+        engine="pandas",
+    )
+    # Pre-processing grpah steps
+    data_set_pandas.graph.to(Imputer(mapping={"department": "RD"}))
+    data_set_pandas._run_db = rundb_mock
+
+    data_set_pandas.reload = unittest.mock.Mock()
+    data_set_pandas.save = unittest.mock.Mock()
+    data_set_pandas.purge_targets = unittest.mock.Mock()
+
+    # Ingest our transactions dataset through our defined pipeline
+    df_pandas = fs.ingest(
+        data_set_pandas,
+        data,
+    )
+
+    assert isinstance(df_pandas, pd.DataFrame)
+
+    # Define the corresponding FeatureSet
+    data_set = fs.FeatureSet(
+        "fs-new",
+        entities=[fs.Entity("id")],
+        description="feature set",
+    )
+    # Pre-processing grpah steps
+    data_set.graph.to(Imputer(mapping={"department": "RD"}))
+    data_set._run_db = rundb_mock
+
+    data_set.reload = unittest.mock.Mock()
+    data_set.save = unittest.mock.Mock()
+    data_set.purge_targets = unittest.mock.Mock()
+
+    # Ingest our transactions dataset through our defined pipeline
```

Comment:
this test section is much more readable than the code it is testing, think about it.